### PR TITLE
Stats Screen implementation and test

### PR DESCRIPTION
**Pull request by NBCCF025**
### Pull Request Description

**Summary**

This pull request introduces a new UI screen and its android test, designed to provide users with a comprehensive overview of their statistics. The UI has been developed to look as the figma design. It is link to the issues #144 and #153.

**Changes Made**
- Implemented the UI of the file named `MyStatsScreen.kt` that is coherent with the figma design.
- Created the android test file named `MyStatsScreenTest.kt`.

**Testing**:
- Implemented the Following tests:
  - **Display test**: Ensures that all the information is displayed correctly.
  - **Scroll test**: Verifies that the box with the letter learned can be scrolled horizontally.
  - **Back Navigation Test**: Confirms that clicking the back button triggers the expected navigation.
- All tests pass successfully, confirming that the screen operates as intended.
- However, I only got 77% test coverage and I have some difficulty to find more test to do for now.

jacocoTestReportMyStatsScreen

### CODE REVIEW

**Comment by NBCCF025** [comment]
As said by the coaches, we can merge this PR to have this task done for sprint 3, and address all the improvement as other task.
This will ensure the PR is not too big and concentrate only on a small part of the app.

**Review by GRGool — APPROVED** [review]
Code is straight forward and well structured. Couple of points I want to mention : 
- I dont see a mention of the change of the utils.kt backbutton. Wish there was a quick explanation in the PR or the related issues.
- You actually have 93.8% coverage if you look at the sonarcloud details. I had a similar issue that I talked about this morning where the percentage displayed is not the correct one(instruction cov/branch cov), good job on the testing. 
- Code duplication is high, mainly comes from the easy/hard/daily/weekly boxes that are very similar so should be easy to factor.
Its good for me, approved

File: app/src/androidTest/java/com/github/se/signify/ui/screens/profile/MyStatsScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    composeTestRule.onNodeWithTag("numberOfDays").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("numberOfDays").assertTextEquals("10 days")
+
+    // Verify letters learned section displays correctly
+    composeTestRule.onNodeWithTag("allLetterLearned").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("allLetterLearned").assertTextEquals("All letters learned")
+    composeTestRule.onNodeWithTag("lettersBox").assertIsDisplayed()
+    lettersLearned.forEach { letter ->
+      composeTestRule.onNodeWithText(letter.toString()).assertIsDisplayed()
+    }
```
Comment (samymessouak) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/profile/MyStatsScreenTest.kt:54]: it ensure each learned letter is displayed as expected, well done

**Review by samymessouak — APPROVED** [review]
**Summary:**
This PR successfully implements `MyStatsScreen`, providing users with a detailed statistics overview that aligns closely with the Figma design. Additionally, the introduction of `MyStatsScreenTest.kt` effectively verifies core functionalities, ensuring that all major components and interactions work as intended.

**Key Points:**

1. **Screen Implementation and Design Consistency:**
   The `MyStatsScreen` UI implementation is clean, intuitive, and visually consistent with the Figma design, offering users a clear presentation of their progress. Great work organizing the layout and defining sections like the top bar, statistics sections, and graphs area. The use of appropriate tags makes the UI more manageable for testing and maintenance.

2. **Testing Strategy and Coverage:**
   The test cases cover important functionality, including:
   - **Display Tests**: Ensuring that all statistics display correctly, verifying tags and values like `numberOfDays`, `lettersBox`, and other key UI elements.
   - **Scroll Functionality**: Confirming the `lettersBox` scrolls horizontally, validating that all learned letters are accessible.
   - **Back Navigation**: Verifying that pressing the `BackButton` triggers the `goBack()` navigation action.
   
   The test coverage of 77% is a strong starting point, and the tests effectively validate the screen’s behavior. Additional tests could be considered in future iterations to address edge cases, such as handling empty lists or varying counts.

3. **Code Structure and Readability:**
   The code is well-organized, with clear separation of concerns, and the naming conventions make it easy to understand the purpose of each component. The navigation setup is efficient and would be straightforward to adapt if additional navigation features are introduced.

**Approval Status:** 
Approved. Overall, this PR provides a solid and user-friendly stats screen with reliable test coverage, meeting the design and functionality requirements effectively.

Great work delivering a polished screen that prioritizes both usability and testability!
